lib/faraday/rack_builder.rb

```diff
@@ -84,6 +84,7 @@ def use(klass, *args, &block)
         use_symbol(Faraday::Middleware, klass, *args, &block)
       else
         raise_if_locked
+        raise_if_adapter_set unless is_adapter?(klass)
```

Comment:
I was unsure whether adapter overwrites should be allowed, so I erred on the side of it being permissive:
https://github.com/lostisland/faraday/pull/685/files#diff-39d210c7f4f77538e42199f60ce1d2f8R151

But I can see how it doesn't make sense, because `#adapter=` twice means one has two adapters. Ideally setting that adapter again should be an overwrite (if permitted), not an additional adapter in the stack.